PatientWebApplication/AppointmentMicroserviceApi/Service/OperationService.cs

```diff
@@ -0,0 +1,242 @@
+        /// </param>
+        /// <returns> List of filtered patient's operations. </returns>
+        private List<Operation> SearchForDate(List<Operation> operations, AppointmentReportSearchDto appointmentSearchDto)
+        {
+            if (!UtilityMethods.CheckIfStringIsEmpty(appointmentSearchDto.Start) && !UtilityMethods.CheckIfStringIsEmpty(appointmentSearchDto.End))
+            {
+                operations = GetOperationsBetweenDates(appointmentSearchDto.Start, appointmentSearchDto.End, operations);
+            }
+            else if (UtilityMethods.CheckIfStringIsEmpty(appointmentSearchDto.Start) && !UtilityMethods.CheckIfStringIsEmpty(appointmentSearchDto.End))
+            {
+                operations = GetOperationsBeforeDate(appointmentSearchDto.End, operations);
+            }
+            else if (!UtilityMethods.CheckIfStringIsEmpty(appointmentSearchDto.Start) && UtilityMethods.CheckIfStringIsEmpty(appointmentSearchDto.End))
+            {
+                operations = GetOperationsAfterDate(appointmentSearchDto.Start, operations);
+            }
+            return operations;
+
+        }
+
+        /// <summary> This method is getting list of filtered <c>Operation</c> of one patient that are between two dates. </summary>
+        /// <param name="operations"><c>operations</c> is List of operations that matches search fields.
+        /// </param>
+        /// <param name="start"><c>start</c> is first date of search.
+        /// </param>
+        /// /// <param name="end"><c>end</c> is last date of search.
+        /// </param>
+        /// <returns> List of filtered patient's operations. </returns>
+        private List<Operation> GetOperationsBetweenDates(string start, string end, List<Operation> operations)
+        {
+            DateTime startDate = UtilityMethods.ParseDateInCorrectFormat(start);
+            DateTime endDate = UtilityMethods.ParseDateInCorrectFormat(end);
+            return operations.FindAll(operation => startDate <= UtilityMethods.ParseDateInCorrectFormat(operation.Date) && UtilityMethods.ParseDateInCorrectFormat(operation.Date) <= endDate);
+        }
+
+        /// <summary> This method is getting list of filtered <c>Operation</c> of one patient that are before given date. </summary>
+        /// <param name="operations"><c>operations</c> is List of operations that matches search fields.
+        /// </param>
+        /// <param name="date"><c>date</c> is last date of search.
+        /// </param>
+        /// <returns> List of filtered patient's operations. </returns>
+        private List<Operation> GetOperationsBeforeDate(string date, List<Operation> operations)
+        {
+            DateTime endDate = UtilityMethods.ParseDateInCorrectFormat(date);
+            return operations.FindAll(operation => UtilityMethods.ParseDateInCorrectFormat(operation.Date) <= endDate);
```

Comment:
Write this in form:
"return operations.FindAll(operation => UtilityMethods.ParseDateInCorrectFormat(operation.Date) <= UtilityMethods.ParseDateInCorrectFormat(date));"